proxy/grpcproxy/chan_stream.go

```diff
@@ -116,17 +116,24 @@ func (s *chanStream) SendMsg(m interface{}) error {
 
 func (s *chanStream) RecvMsg(m interface{}) error {
 	v := m.(*interface{})
-	select {
-	case msg, ok := <-s.recvc:
-		if !ok {
-			return grpc.ErrClientConnClosing
+	for {
+		select {
+		case msg, ok := <-s.recvc:
+			if !ok {
+				s.cancel()
```

Comment:
This makes sense.
Did we just forget to cancel this context before? And now it fixes?